packages/mobile/src/send/utils.ts

```diff
@@ -188,34 +184,32 @@ export function* handleSendPaymentData(
   }
   yield put(storeLatestInRecents(recipient))
 
-  if (data.currencyCode) {
-    yield put(selectPreferredCurrency(data.currencyCode as LocalCurrencyCode))
-    const action: FetchCurrentRateSuccessAction | FetchCurrentRateFailureAction = yield take([
-      Actions.FETCH_CURRENT_RATE_SUCCESS,
-      Actions.FETCH_CURRENT_RATE_FAILURE,
-    ])
-    if (action.type === Actions.FETCH_CURRENT_RATE_FAILURE) {
-      yield put(showError(ErrorMessages.EXCHANGE_RATE_FAILED))
-      Logger.warn(TAG, '@handleSendPaymentData failed to fetch current rate')
-      return
-    }
-  }
-
-  if (data.amount) {
+  if (data.amount && data.currencyCode) {
     // TODO: integrate with SendConfirmation component
-    const exchangeRate = yield select(getLocalCurrencyExchangeRate)
-    const amount = convertLocalAmountToDollars(data.amount, exchangeRate)
-    if (!amount) {
+    // const exchangeRate = yield select(getLocalCurrencyExchangeRate)
+    const currency = data.currencyCode as LocalCurrencyCode
+    const exchangeRate = yield call(fetchExchangeRate, currency)
```

Comment:
Can you please define the type of `exchangeRate`? I believe it's a string but `convertLocalAmountToDollars` expects a BigNumber

Reply:
+1, note that we don't have automatic types for sagas so being explicit with `const exchangeRate: string = yield call(fetchExchangeRate, currency)` is needed to type check